### PR TITLE
fix: Add keys to list elements

### PR DESCRIPTION
**Pull request by sarangj**
## Ticket:

- JIRA ticket [DR-xxxx](url)

## This PR does the following:

Add keys to elements in the metadata and citations that are structured in a list.  Note I had to replace a couple instances of `<>` fragment syntax in favor of the explicit `<Fragment` to pass in a key prop.

This fixes *most* warnings for missing / duplicate keys - there are some edge cases I've seen that are a bit tricky for me, and would have to take more investigation

## How has this been tested? How should a reviewer test this?

Opened an item page locally and verified the warnings go away
### Checklist:

- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.
- [ ] I have updated the CHANGELOG.md.

### CODE REVIEW

File: app/src/components/items/overview/citations/overview.tsx [hunk 1]
```diff
@@ -90,14 +91,12 @@ ${metadata.origin ? "(" + metadata.origin + ")" : ""}${
           const value = citationFormatDummyData[field];
           if (value !== "") {
             return (
-              <>
-                <Text key={index} size="overline1" marginBottom="xs">
+              <Fragment key={`citation-${field}`}>
```
Comment (avertrees) [inline app/src/components/items/overview/citations/overview.tsx:94]: curious about why this needs a Fragment?

Comment (avertrees) [reply app/src/components/items/overview/citations/overview.tsx:94]: I'll read the docs

**Review by avertrees — APPROVED** [review]

File: app/src/components/items/overview/citations/overview.tsx (hunk 1 above)
Comment (sarangj) [reply app/src/components/items/overview/citations/overview.tsx:94]: `<>` is just shorthand for a `Fragment` - if you want to add any props to a `<>` though, you have to explicitly use `Fragment`
